src/main/kotlin/com/y9san9/kotlogram/storage/ApiStorage.kt

```diff
@@ -4,34 +4,34 @@ import com.github.badoualy.telegram.api.TelegramApiStorage
 import com.github.badoualy.telegram.mtproto.auth.AuthKey
 import com.github.badoualy.telegram.mtproto.model.DataCenter
 import com.github.badoualy.telegram.mtproto.model.MTSession
-import com.y9san9.kotlogram.utils.refresh
-import java.io.File
-import java.io.FileNotFoundException
+import com.y9san9.kds.KDataStorage
+import com.y9san9.kds.commit
 
+private object Storage : KDataStorage() {
+    var authKey by property<ByteArray>()
+    var dataCenter by property<String>()
+    var session by property<MTSession>()
+}
 
+//TODO: Remove constructor parameter or add creating custom path feature to kds
 class ApiStorage(name: String = "") : TelegramApiStorage {
-    private val authKeyFile = File(System.getProperty("user.dir"),"properties/auth$${name}.key")
-    private val nearestDCFile = File(System.getProperty("user.dir"),"properties/dc$${name}.save")
+    /*private val authKeyFile = File(System.getProperty("user.dir"), "properties/auth$${name}.key")
+    private val nearestDCFile = File(System.getProperty("user.dir"), "properties/dc$${name}.save")*/
 
-    override fun saveAuthKey(authKey: AuthKey) = authKeyFile.apply { refresh() }.writeBytes(authKey.key)
-    override fun loadAuthKey() = try {
-        AuthKey(authKeyFile.readBytes())
-    } catch (_: FileNotFoundException){
-        null
-    }
+    override fun saveAuthKey(authKey: AuthKey) = Storage.commit { this.authKey = authKey.key }
+    override fun loadAuthKey() = AuthKey(Storage.authKey)
 
-    override fun saveDc(dataCenter: DataCenter) = nearestDCFile.apply { refresh() }.writeText(dataCenter.toString())
-    override fun loadDc() = try {
-        nearestDCFile.readText().split(":").let { (ip, port) ->
-            DataCenter(ip, port.toInt())
-        }
-    } catch (_: FileNotFoundException) {
-        null
+    override fun saveDc(dataCenter: DataCenter) = Storage.commit { this.dataCenter = dataCenter.toString() }
+    override fun loadDc() = Storage.dataCenter.split(":").let { (ip, port) ->
+        DataCenter(ip, port.toInt())
     }
 
-    override fun deleteAuthKey() = authKeyFile.deleteRecursively().let {  }
-    override fun deleteDc() = nearestDCFile.deleteRecursively().let {  }
+    //TODO: Create delete function for kds
+    override fun deleteAuthKey() { }
+    override fun deleteDc() { }
 
-    override fun saveSession(session: MTSession?) {}
-    override fun loadSession(): MTSession? = null
+    override fun saveSession(session: MTSession?) = Storage.commit {
+        this.session = session!!
+    }
+    override fun loadSession(): MTSession? = Storage.session
 }
```

Comment:
Good job! But for multi-session it'll be failed cuz of object Storage. 
Solution: move `object Storage` inside `ApiStorage` class (also it should be private), and make something like this:
```kotlin
private object Storage : KDataStorage(File(System.getProperty("user.dir"), name))
```

Reply:
also, please, specify version of "kotlin-data-storage" explicitly, cuz it can be changed any time